grafana-alloy.yaml

```diff
@@ -26,6 +29,9 @@ pipeline:
     with:
       deps: github.com/hashicorp/go-retryablehttp@v0.7.7 google.golang.org/grpc@v1.64.1
 
+  - name: Generate UI
+    runs: make generate-ui
```

Comment:
Where do these files end up @JonahEdick? I don't see anything in the final packages.

The build log shows this:

```
2024/07/30 19:21:26 INFO Compiled successfully.
2024/07/30 19:21:26 INFO
2024/07/30 19:21:26 INFO File sizes after gzip:
2024/07/30 19:21:26 INFO
2024/07/30 19:21:26 INFO   409.57 kB  build/static/js/main.cba49e5e.js
2024/07/30 19:21:26 INFO   4.25 kB    build/static/css/main.314ea319.css
2024/07/30 19:21:26 INFO
2024/07/30 19:21:26 INFO The project was built assuming it is hosted at ./public/.
2024/07/30 19:21:26 INFO You can control this with the homepage field in your package.json.
2024/07/30 19:21:26 INFO
2024/07/30 19:21:26 INFO The build folder is ready to be deployed.
```

But when I install the package I don't see any of those files. Are they compiled into the binary, or should we have a subpackage for the UI?

Reply:
According to Grafana's docker file, it states the following
```
# Build the UI before building Alloy, which will then bake the final UI into
# the binary.
```
So by that logic I would assume that there isn't any extra files because they get put into the same grafana-alloy binary that we were building before

Reply:
https://github.com/grafana/alloy/blob/f439621443311d1c293f16162f35083c8d45cbc3/internal/web/ui/assets_builtin.go#L15 here!

Reply:
When I tried installing our package and running the UI I get a 404 page. But running the image from their Docs shows it at 127.0.0.1:12345

https://grafana.com/docs/alloy/latest/set-up/install/docker/

So I think we need to figure out how to get the yarn/npm assets into the right place for go build to pick them up.

Reply:
Found: https://github.com/grafana/alloy/blob/b068c13bab9ff8f6351c68096a9e18b58acda1eb/Dockerfile#L29C4-L29C61

TLDR;

_GO_TAGS="netgo builtinassets promtail_journal_enabled"_

Reply:
How would one go at adding tags to the go/build process on the pipeline?

Reply:
https://github.com/chainguard-dev/melange/blob/5beea3ebd2162328f01d756674091797e8af88d1/pkg/build/pipelines/go/build.yaml#L22

Reply:
good news! 
```
/work/packages # wget localhost:12345
--2024-07-31 19:17:49--  http://localhost:12345/
Resolving localhost... ::1, 127.0.0.1
Connecting to localhost|::1|:12345... failed: Connection refused.
Connecting to localhost|127.0.0.1|:12345... connected.
HTTP request sent, awaiting response... 200 OK
Length: 707 [text/html]
index.html: Read-only file system
```
I was able to retrieve the webpage on the local-wolfi container running grafana-alloy! (I can verify that it is the expected outcome after making an image with it)

Reply:
I made some changes on the PR could you test it if that works as you expected? 

Also tried to bump the docker deps but the compilation failed with:

```
2024/07/31 19:44:08 INFO Done in 39.15s.
2024/07/31 19:44:08 INFO running step "Run a build using the go compiler"
2024/07/31 19:44:55 WARN # github.com/google/cadvisor/container/docker/utils
2024/07/31 19:44:55 WARN /var/cache/melange/gomodcache/github.com/grafana/cadvisor@v0.0.0-20231110094609-5f7917925dea/container/docker/utils/docker.go:48:42: undefined: dockertypes.Info
2024/07/31 19:44:55 WARN /var/cache/melange/gomodcache/github.com/grafana/cadvisor@v0.0.0-20231110094609-5f7917925dea/container/docker/utils/docker.go:57:44: undefined: dockertypes.Info
2024/07/31 19:44:55 WARN /var/cache/melange/gomodcache/github.com/grafana/cadvisor@v0.0.0-20231110094609-5f7917925dea/container/docker/utils/docker.go:77:43: undefined: dockertypes.Info
2024/07/31 19:44:55 WARN /var/cache/melange/gomodcache/github.com/grafana/cadvisor@v0.0.0-20231110094609-5f7917925dea/container/docker/utils/docker.go:86:48: undefined: dockertypes.ImageSummary
```

which is good because we could use that information while filling the note of the advisory data.

Reply:
Yeah I can pull and test this. One second

Reply:
Looks good now, I can see the UI!